src/m365/spo/commands/folder/folder-add.spec.ts

```diff
@@ -194,6 +197,44 @@ describe(commands.FOLDER_ADD, () => {
     });
   });
 
+  it('creates folder and ensures parent folder path', async () => {
+    const requestGet = sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf('GetFolderByServerRelativePath(') > -1) {
+        return {
+          Exists: false
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    const requestPost: sinon.SinonStub = stubPostResponses();
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        parentFolderUrl: '/Shared Documents/abc',
+        name: 'folder1',
+        ensureParentFolders: true
+      }
+    });
+
+    assert.strictEqual(requestGet.lastCall.args[0].url, 'https://contoso.sharepoint.com/_api/web/GetFolderByServerRelativePath(DecodedUrl=\'Shared%20Documents%2Fabc\')?$select=Exists');
+
+    assert(requestPost.calledWith({
+      url: `https://contoso.sharepoint.com/_api/web/folders/addUsingPath(decodedUrl='${formatting.encodeQueryParameter('Shared Documents/abc')}')`,
+      headers: { accept: 'application/json;odata=nometadata' },
+      responseType: 'json'
+    }));
+
+    assert(requestPost.calledWith({
+      url: `https://contoso.sharepoint.com/_api/web/folders/addUsingPath(decodedUrl='${formatting.encodeQueryParameter('/Shared Documents/abc/folder1')}')`,
+      headers: { accept: 'application/json;odata=nometadata' },
+      responseType: 'json'
+    }));
```

Comment:
Instead of doing this, let's write the sinon.stub(request, 'post') in this test ourselves, check on the specific url and at the end check on the amount of times the postStub got called.